File: test2.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	const LENGTH int64 = 1_000_000_000 
	arr := make([]int32, LENGTH)      

	for i := int64(0); i < LENGTH; i++ {
		arr[i] = int32(i)
	}
	var sum int64
	for _, n := range arr {
		sum += int64(n)
	}

	fmt.Println("sum:", sum)
	fmt.Println("time required:", time.Since(start))
} // ChatGpt 

File: test2.rs
use std::time::Instant;

const LEN: usize = 1_000_000_001;

fn heap_stuff() -> i64 {
    let mut arr: Vec<i32> = Vec::with_capacity(LEN);

    for i in 0..LEN {
        arr.push(i as i32);
    }

    let mut result: i64 = 0;
    for &i in &arr {
        result += i as i64;
    }

    result
}

fn main() {
    let start = Instant::now();
    let result = heap_stuff();
    let elapsed = start.elapsed();

    println!("Result: {}", result);
    println!("Time required: {:?}", elapsed);
} // ChatGpt 